test/TestServerProjectsLowLevel/body-complex/Generated/PrimitiveRestClient.cs

```diff
@@ -14,849 +14,1802 @@
+        public PrimitiveRestClient(AzureKeyCredential credential, Uri endpoint = null, AutoRestComplexTestServiceClientOptions options = null)
         {
-            var message = _pipeline.CreateMessage();
-            var request = message.Request;
-            request.Method = RequestMethod.Get;
-            var uri = new RawRequestUriBuilder();
-            uri.Reset(endpoint);
-            uri.AppendPath("/complex/primitive/integer", false);
-            request.Uri = uri;
-            request.Headers.Add("Accept", "application/json");
-            return message;
+            if (credential == null)
+            {
+                throw new ArgumentNullException(nameof(credential));
+            }
+            endpoint ??= new Uri("http://localhost:3000");
+
+            options ??= new AutoRestComplexTestServiceClientOptions();
+
+            _clientDiagnostics = new ClientDiagnostics(options);
+            _keyCredential = credential;
+            _pipeline = HttpPipelineBuilder.Build(options, new HttpPipelinePolicy[] { new LowLevelCallbackPolicy() }, new HttpPipelinePolicy[] { new AzureKeyCredentialPolicy(_keyCredential, AuthorizationHeader) }, new ResponseClassifier());
+            _endpoint = endpoint;
         }
 
         /// <summary> Get complex types with integer properties. </summary>
         /// <param name="options"> The request options. </param>
-        public async Task<Response> GetIntAsync(RequestOptions options = null)
-        {
-            options ??= new RequestOptions();
-            using HttpMessage message = CreateGetIntRequest();
-            RequestOptions.Apply(options, message);
-            await _pipeline.SendAsync(message, options.CancellationToken).ConfigureAwait(false);
-            if (options.StatusOption == ResponseStatusOption.Default)
-            {
-                switch (message.Response.Status)
-                {
-                    case 200:
-                        return message.Response;
-                    default:
-                        throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
-                }
-            }
-            else
-            {
-                return message.Response;
+        /// <remarks>
+        /// Schema for <c>Response Body</c>:
+        /// <code>{
+        ///   field1: number,
+        ///   field2: number
+        /// }
+        /// </code>
+        /// Schema for <c>Response Error</c>:
+        /// <code>{
+        ///   status: number,
+        ///   message: string
+        /// }
+        /// </code>
+        /// 
+        /// </remarks>
+#pragma warning disable AZC0002
+        public virtual async Task<Response> GetIntAsync(RequestOptions options = null)
+#pragma warning restore AZC0002
+        {
+            using var scope = _clientDiagnostics.CreateScope("PrimitiveRestClient.GetInt");
```

Comment:
This also doesn't allow client authors to add more attributes to the scope.

Reply:
Hm... there may be no wrapping method. Should we add a property to `RequestOptions`?

Reply:
There are two scenarios that might be conflicted here. There might be a path to unify them but I'd like to understand it first before we shake the world.

1. We generate internal rest clients to be wrapped by data-plane T2 libraries. 
2. We generate public LLC clients

For the first case, we don't know what scopes need to be created and what attributes are attached. It's possible that multiple rest calls would share a single diagnostic scope.

For the second case, we know the scope name is correct because the method is public.

Reply:
>Should we add a property to RequestOptions?

That doesn't work. `ClientDiagnostics` is designed in a way where it's completely 0-cost when no one is listening. Forcing the caller to always allocate RequestOptions would change that.

Reply:
This whole change is LLC-only scope